### DISCUSSION TITLE
[Feature Request] Header and Footer templates for Releases

### DISCUSSION THREAD

**Discussion by hawkeye64** [community/community#6936] [Other Feature Feedback, Questions, & Ideas] [answered]
Previously brought up the discussion here: https://github.com/githubevents/universe2021/discussions/228#discussioncomment-1554284

---

I think it'd be awesome if we could create templates for a Header and Footer for Releases.

For instance, whatever you type would show up for the header and footer. But, even better is variables.
Let's say I could do something like this for the Header:

```
# ${packageName} v${packageVersion} released!
```

Then the **name** and **version** from `package.json` (or other similar for other languages).

Then, if package.name was "My Super Package" and package.version was "1.0.0", then the markdown in the Release would look like this:

```md
# My Super Package v1.0.0 released!
```
Which would result in:

# My Super Package v1.0.0 released!

I think this would really be useful. As for the footer, I could always put in my GitHub sponsorship link. But, there are so many possibilities here.

-Jeff

**Comment by MylesBorins** [comment, marked as answer]
👋  @hawkeye64 thanks for the suggestion

At this time this is not something we are going to be exploring. One of the design constraints we took on for configuration was not introducing variables at all. Deciding on syntax, where to source content from significantly increases the scope of what we built (and leaves less design room in the future).

While it is not as streamlined as supporting this functionality directly we do have an API you can use to get the release notes.

https://docs.github.com/en/rest/reference/repos#generate-release-notes-content-for-a-release

You can use a GitHub Action Workflow and the GitHub Script action to call the API, get the content, and prepend / append additional content programatically. This obviously is not nearly as nice as having it right in the config and pressing a button, but at least it is possible 🎉 

I'm going to mark this as answered as there isn't additional work items to do for the moment, but please feel free to ask any questions.
